lib/services/livesync/playground/devices/preview-devices-service.ts

```diff
@@ -0,0 +1,36 @@
+import { Device } from "nativescript-preview-sdk";
+import { EventEmitter } from "events";
+import { DeviceDiscoveryEventNames } from "../../../../common/constants";
+
+export class PreviewDevicesService extends EventEmitter implements IPreviewDevicesService {
+	public connectedDevices: Device[] = [];
+
+	public onDevicesPresence(devices: Device[]): void {
+		_(devices)
+			.reject(d => _.find(this.connectedDevices, device => d.id === device.id))
+			.each(device => this.raiseDeviceFound(device));
+
+		_(this.connectedDevices)
+			.reject(d => _.find(devices, device => d.id === device.id))
+			.each(device => this.raiseDeviceLost(device));
+	}
+
+	public getDeviceById(id: string): Device {
+		return _.find(this.connectedDevices, { id });
+	}
+
+	public getDevicesForPlatform(platform: string): Device[] {
+		return _.filter(this.connectedDevices, { platform: platform.toLowerCase() });
+	}
+
+	private raiseDeviceFound(device: Device) {
+		this.emit(DeviceDiscoveryEventNames.DEVICE_FOUND, device);
```

Comment:
Can you add a test case that the event is emitted. Same for `DeviceDiscoveryEventNames.DEVICE_LOST`.